compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -399,3 +419,90 @@ checkDFlags Options {..} dflags@DynFlags {..}
                 " imports a package with the same name. \
             \ Please check your dependencies and rename the package you are compiling \
             \ or the dependency."
+
+-- Expand SDK package dependencies using the SDK root path.
+-- E.g. `daml-trigger` --> `$DAML_SDK/daml-libs/daml-trigger.dar`
+-- When invoked outside of the SDK, we will only error out
+-- if there is actually an SDK package so that
+-- When there is no SDK
+expandSdkPackages :: [FilePath] -> IO [FilePath]
+expandSdkPackages dars = do
+    mbSdkPath <- handleIO (\_ -> pure Nothing) $ Just <$> getSdkPath
+    mapM (expand mbSdkPath) dars
+  where
+    isSdkPackage fp = takeExtension fp `notElem` [".dar", ".dalf"]
+    expand mbSdkPath fp
+      | fp `elem` basePackages = pure fp
+      | isSdkPackage fp = case mbSdkPath of
+            Just sdkPath -> pure $ sdkPath </> "daml-libs" </> fp <.> "dar"
+            Nothing -> fail $ "Cannot resolve SDK dependency '" ++ fp ++ "'. Use daml assistant."
+      | otherwise = pure fp
+
+
+mkPackageFlag :: UnitId -> PackageFlag
+mkPackageFlag unitId = ExposePackage "--package " (UnitIdArg unitId) (ModRenaming True [])
+
+mkBaseUnits :: Maybe String -> [UnitId]
+mkBaseUnits optMbPackageName
+  | optMbPackageName == Just "daml-prim" =
+      []
+  | optMbPackageName == Just damlStdlib =
+      [ stringToUnitId "daml-prim" ]
+  | otherwise =
+      [ stringToUnitId "daml-prim"
+      , stringToUnitId damlStdlib ]
+
+dependantUnitsFromDamlYaml :: Maybe FilePath -> IO [UnitId]
+dependantUnitsFromDamlYaml mbRoot = do
+  let dir = fromMaybe "." mbRoot
+  withCurrentDirectory dir $ do
```

Comment:
Why do we have to change the directory here? Is that only so that `calcUnitsFromDeps` resolves relative file paths directly? If so, I would prefer to instead pass on the directory to `calcUnitsFromDeps`. CWD is process-global state so modifying it is a bit fishy and will almost certainly break in some cases without a lock.

Reply:
ok, `withCurrentDirectory` is removed